### ISSUE TITLE
[new resource]: Methodological Workshop - Computational Background Skills for Digital Humanities University of Vienna - Digital Humanities

### ISSUE THREAD

**Issue by emilykateemilykate** [DARIAH-ERIC/dariah-campus#781]
### Title of the resource

Methodological Workshop - Computational Background Skills for Digital Humanities University of Vienna - Digital Humanities

### Resource type

Hosted Resource

### Authors, editors and contributors

Dr. Tara Andrews, Emily Genatowski

### Topics (keywords)

Computational Skills

### Learning outcomes

This course is intended to provide basic training and support for further skills courses in the Digital Humanities. It is strongly recommended as a prerequisite to the other DH practical courses. 

Introduction to the command line
Introduction to file formats and filesystems
Solving the mystery of special characters
Understanding how the Internet works behind the browser
Things you ought to know about your (Mac / Windows / Linux) operating system
Where to go for help and how to understand the answers

### Abstract

Day 1
Install needed software to be used as terminal and code editor
Start to become familiar with typing commands at the prompt of your computer's shell
Make a note of all the error and informational messages you encounter along the way
Start to become familiar with how problems are communicated by computer programs
Day 2
Understand what regular expressions are for and when and why they are useful
Learn the basic syntax of regular expressions and how to construct match patterns
Day 3
Learn how to manipulate the contents of files on the command line
Understand the concept of file types and formats, and understand how your computer guesses what kind of file something is
Learn the file command, which takes a different approach to guessing what kind of file something is
Practice getting information out of the error messages you have collected so far
Day 4
Understand what happens behind the scenes when you navigate to a URL in a web browser, or when you send an email
Make requests of Web servers the same way that browsers do
Learn about the three "Web languages" (HTML, CSS, Javascript) and how they work together to define the content, look, and functionality of web pages
Day 5
Create an empty repository on Github with a README and an MIT Licence
Practice working with Git and Github, the add/commit/push cycle, and pulling remote changes
Understand how code collaboration works on Github, and fork the miniproject repository
Use regular expression knowledge to collect, transform, and clean data taken from Wikipedia into a format suitable for a web application
Master the terminology that programmers take for granted
Day 6
Write simple routines on the command line
Learn how to turn these routines into executable programs
Learn about different sorts of shells
Modify the JavaScript function given in the boilerplate code to execute various regex tasks
See the output of regex by opening the index.html file with a browser
Understand how our code fetches the data using HTTP requests
Add and modify basic CSS styles for different elements
Add and modify basic interaction elements e.g. buttons in HTML
Bind the elements to our JavaScript functions to toggle the data being displayed
Finalize the project, push it to git repo and make it online using GitHub Pages

**Comment by VickyGarnett** [comment]
Hi @emilykateemilykate thanks for this contribution proposal.  I wonder if we could find time to have a quick chat about this at some point in the next couple of days so I can get a clearer handle on what you aim to produce.  It's possible from what you describe that this might be better as a 'Captured Event' rather than a 'Hosted Resource'.  I can also then give you a quick tour of the CMS.

Thanks

V

**Comment by emilykateemilykate** [comment]
Yes, please!! Perhaps we can talk sometime tomorrow? I can likely make any
time work over zoom! Thank you so so much for your feedback and willingness
to showcase the cms! I can’t wait to get started.

Emily Kate

**Comment by VickyGarnett** [comment]
Great, OK, assuming you're the Emily Kate that Vera Charvat has just included in a different email conversation, then I will make arrangements with you there :)

**Comment by emilykateemilykate** [comment]
Hello!

I have been thinking about this resource and how we may want to tackle this correctly.  I am actually taking the course this semester in order to better understand the way in which it should be hosted on the platform and I actually think as opposed to being a hosted resource, it may do better as an external resource.  Much of the technical aspect of the course is drawn from datacamp and Software carpentry's materials and as such, I think perhaps we may do better to express it as an external resource.  I'm open to thoughts on this!

**Comment by VickyGarnett** [comment]
Hi @emilykateemilykate - I don't think this resource got drafted up, or did it get transformed into something else?  I'm doing a clean up of issues, and don't want to close down any active ones :). Thanks. V

**Comment by emilykateemilykate** [comment]
I believe this is no longer active!
